nucypher-core-wasm/src/lib.rs

```diff
@@ -648,6 +653,167 @@ impl SessionSecretFactory {
+        public_key: &DkgPublicKey,
+        conditions: &OptionConditions,
+    ) -> Result<AuthenticatedData, Error> {
+        let typed_conditions = try_from_js_option::<Conditions>(conditions)?;
+
+        Ok(Self(nucypher_core::AuthenticatedData::new(
+            public_key.as_ref(),
+            typed_conditions.as_ref().map(|conditions| &conditions.0),
+        )))
+    }
+
+    pub fn aad(&self) -> Box<[u8]> {
+        self.0.aad()
+    }
+
+    #[wasm_bindgen(getter, js_name = publicKey)]
+    pub fn public_key(&self) -> DkgPublicKey {
+        DkgPublicKey::from(self.0.public_key)
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn conditions(&self) -> Option<Conditions> {
+        self.0.conditions.clone().map(Conditions)
+    }
+
+    #[wasm_bindgen(js_name = toBytes)]
+    pub fn to_bytes(&self) -> Box<[u8]> {
+        to_bytes(self)
+    }
+}
+
+//
+// Encrypt for dkg
+//
+#[wasm_bindgen(js_name = "encryptForDkg")]
+pub fn encrypt_for_dkg(
+    data: &[u8],
+    public_key: &DkgPublicKey,
+    conditions: &OptionConditions,
+) -> Result<DkgEncryptionResult, Error> {
+    let typed_conditions = try_from_js_option::<Conditions>(conditions)?;
+    let (ciphertext, auth_data) = nucypher_core::encrypt_for_dkg(
+        data,
+        public_key.as_ref(),
+        typed_conditions.as_ref().map(|conditions| &conditions.0),
+    )
+    .map_err(map_js_err)?;
+    Ok(into_js_array([
+        JsValue::from(Ciphertext::from(ciphertext)),
+        JsValue::from(AuthenticatedData::from(auth_data)),
+    ]))
+}
+
+//
+// AccessControlPolicy
+//
+
+#[wasm_bindgen]
+#[derive(PartialEq, Eq, Debug, derive_more::From, derive_more::AsRef)]
+pub struct AccessControlPolicy(nucypher_core::AccessControlPolicy);
+
+generate_from_bytes!(AccessControlPolicy);
+generate_equals!(AccessControlPolicy);
+
+#[wasm_bindgen]
+impl AccessControlPolicy {
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        auth_data: &AuthenticatedData,
+        authorization: &[u8],
+    ) -> Result<AccessControlPolicy, Error> {
+        Ok(Self(nucypher_core::AccessControlPolicy::new(
+            auth_data.as_ref(),
+            authorization,
+        )))
+    }
+
+    pub fn aad(&self) -> Box<[u8]> {
+        self.0.aad()
+    }
+
+    #[wasm_bindgen(getter, js_name = publicKey)]
+    pub fn public_key(&self) -> DkgPublicKey {
+        DkgPublicKey::from(self.0.auth_data.public_key)
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn authorization(&self) -> Box<[u8]> {
+        self.0.authorization.clone()
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn conditions(&self) -> Option<Conditions> {
+        self.0.auth_data.conditions.clone().map(Conditions)
+    }
+
+    #[wasm_bindgen(js_name = toBytes)]
+    pub fn to_bytes(&self) -> Box<[u8]> {
+        to_bytes(self)
+    }
+}
+
+//
+// ThresholdMessageKit
+//
+#[wasm_bindgen]
+#[derive(PartialEq, Eq, Debug, derive_more::From, derive_more::AsRef)]
+pub struct ThresholdMessageKit(nucypher_core::ThresholdMessageKit);
```

Comment:
I've noticed that in some places in `nucypher-core`, we're using `{ backend: InnerType }` and `(InnerType)` in others. Perhaps we could settle on one way to define structs. It's not relevant to this PR, just a general comment on style